In [2]:
import pandas as pd
import scipy 
import numpy as np
from scipy import stats

# Hypothesis testing

### Hypothesis Statement

- Null Hypothesis (H₀): The weekly operating cost has not increased. That is, the mean weekly cost is **μ = $1,000 + $5 × 600 = $4,000**.
- Alternative Hypothesis (H₁): The weekly operating cost has **increased**, i.e., the mean weekly cost is **greater than $4,000**.

### Calculate the Test Statistic

In [3]:
mean_weekly_cost = 1000+5*600 #The theoretical mean weekly cost μ
deviation_weekly_cost = 5*25 #Standard deviation of weekly cost σ
n = 25 #Sample size
x = 3050 #Sample mean

In [4]:
#Test statistic Z
z = (x-mean_weekly_cost)/(deviation_weekly_cost/np.sqrt(n))
z

-38.0

### Determine the Critical Value

In [5]:
z_critical = stats.norm.ppf(1-0.05)
z_critical

1.644853626951472

### Make a Decision
Since the test statistic Z = -38.00 is not greater than the critical value Z_critical = 1.64, we fail to reject the null hypothesis.


### Conclusion

There is not sufficient evidence at the 5% significance level to support the restaurant owners' claim that the weekly operating costs are higher than the model suggests.


# Chi-Square

### Hypthesis statemenets
- Null Hypothesis (H₀): There is **no association** between device type and customer satisfaction.
- Alternative Hypothesis (H₁): There **is an association** between device type and customer satisfaction.

### Compute Chi-Square Statistic

In [6]:
# Observed frequencies (from the contingency table)
observed = pd.DataFrame({
    "Smart Thermostat": [50, 80, 60, 30, 20],
    "Smart Light": [70, 100, 90, 50, 50]
}, index=["Very Satisfied", "Satisfied", "Neutral", "Unsatisfied", "Very Unsatisfied"])

In [7]:
# Run the chi-square test
chi2_stat, p_val, dof, expected = scipy.stats.chi2_contingency(observed)

In [8]:
print(f"Chi-Square Statistic: {chi2_stat:.2f}")
print(f"Degrees of Freedom: {dof}")
print(f"P-value: {p_val:.4f}")

# Show expected frequencies in a readable format
expected_df = pd.DataFrame(expected, index=observed.index, columns=observed.columns)
print("\nExpected Frequencies Table:")
print(expected_df.round(2))


Chi-Square Statistic: 5.64
Degrees of Freedom: 4
P-value: 0.2278

Expected Frequencies Table:
                  Smart Thermostat  Smart Light
Very Satisfied                48.0         72.0
Satisfied                     72.0        108.0
Neutral                       60.0         90.0
Unsatisfied                   32.0         48.0
Very Unsatisfied              28.0         42.0


### Determine the Critical Value

In [11]:
# Degrees of freedom = (rows - 1) * (columns - 1)
dof = 4

# Critical value from chi-square distribution
critical_value = stats.chi2.ppf(1 - 0.05, dof)
critical_value

9.487729036781154

### Make a decision
Since the Chi-Square Statistic = 5.64 is not greater than the critical value = 9.49, we fail to reject the null hypothesis.


### Conclusion

There is not sufficient evidence to conclude that customer satisfaction is associated with the type of smart home device purchased.
